library/Fission/Web/Auth/Error.hs

```diff
@@ -0,0 +1,21 @@
+module Fission.Web.Auth.Error (Error (..)) where
+
+import Fission.Prelude
+
+data Error
+  = NoToken
+  | BadToken
+  | NoSuchUser
+  | Unauthorized
+  deriving ( Exception
+           , Eq
+           , Generic
+           , ToJSON
+           )
+
+instance Show Error where
+  show = \case
+    NoToken      -> "No token included on request"
+    BadToken     -> "Token is improperly formatted"
+    NoSuchUser   -> "No such user exists"
+    Unauthorized -> "User not authorized"
```

Comment:
Why `Show` instead of `Display`?

Reply:
tbh haven't really figured out the difference. Why `Display` instead of `Show`?

Reply:
`Display` is way more efficient than `Show`, and works really nicely with the `log*` functions. The problem with `Show` is that it generates a `String` (i.e. `[Char]`), which is a simple data structure, but allocates a bunch of thunks (memory), is O(n) to traverse, and appending is O(n+m). `Display` uses a text builder, which is much more efficient in memory and compute.

You can get both pretty easily:

```haskell
instance Display Error where
  display = \case
    ...

instance Show Error where
  show = show . display
```

Reply:
oh sweet makes sense :+1: